runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterPlugin.java

```diff
@@ -183,10 +185,22 @@ boolean shouldFilterPlayerMessage(String playerName)
 			(config.filterClan() || !clanManager.isClanMember(playerName));
 	}
 
-	String censorMessage(final String message)
+	String censorMessage(final String username, final String message)
 	{
 		String strippedMessage = jagexPrintableCharMatcher.retainFrom(message)
 			.replace('\u00A0', ' ');
+		if (shouldFilterByName(username))
+		{
+			switch (config.filterType())
+			{
+				case CENSOR_WORDS:
+					return StringUtils.repeat('*', strippedMessage.length());
```

Comment:
Move the `CENSOR_WORDS` code to a separate method and use it here and in the `filteredPatterns` loop.

Reply:
I think this is fine as-is